### DISCUSSION TITLE
Add progress option to Vtracer

### DISCUSSION THREAD

**Discussion by zamazan4ik** [visioncortex/vtracer#68] [Ideas]
Hi!

I found that Vtracer performs conversion too slowly on this sample: https://sample-videos.com/download-sample-jpg-image.php , 30 Mib image with the following command: `vtracer --input input_30mib.jpg --output output.svg`. Vtracer is built locally with `cargo build --release` with Rust 1.74 on the latest `master` branch, commit `74f2a04a17d8c246d80c439fb162780160a7c3e9`. It took `2341.102 s [User: 2291.374 s, System: 42.931 s]` to convert the image (according to Hyperfine) even on my quite good AMD Ryzen 5900x with 48 Gib RAM (no swapping) I guess it's not an internal bug - just the input image is quite large to convert. 

From the user perspective, it would be nice to have an option to track the conversion progress somehow. So I will be able to estimate the finish time and check that Vtracker doesn't stuck and works properly.

**Comment by tyt2y3** [comment]
What's the resolution of the image? The pipeline is not parallelized, so it is bound to 1 CPU core.

**Reply by zamazan4ik** [reply]
The image resolution is 13583  x  5417 (quite large, yeah). Yes, I see that Vtracer is bound to 1 CPU. I am not complaining a lot about the conversion speed (however having a faster conversion is always better) - I just want to have the ability to track the progress of the conversion process.

**Comment by tyt2y3** [comment]
73MP is not that bad. If you tile the image, it can easily be parallalized. The web GUI has a progress bar, may be you can take a look into how it calculates the progress and port the bits into the CLI.
